master/internal/db/postgres_experiments.go

```diff
@@ -555,12 +521,10 @@ WHERE t.id=$2
   AND v.metrics->'validation_metrics'->$1 IS NOT NULL
 ORDER BY batches;`, metricName, trialID, startBatches, endBatches, startTime)
 	if err != nil {
-		return nil, nil, maxEndTime, errors.Wrapf(err, "failed to get metrics to sample for experiment")
+		defer rows.Close()
+		return nil, errors.Wrapf(err, "failed to get metrics to sample for experiment")
 	}
-	defer rows.Close()
```

Comment:
it looks like we are relying on the function that calls this to do `rows.Close()` on the rows that are returned, in the case that there is not an error? 

it seems safer to have it where this function unconditionally defers the closing of the rows itself.

Reply:
This function was refactored to split the DB query from specific business login the API layer. The issue here is by returning rows. If they are `closed` in this function the results cannot be read elsewhere which is why this code is written as is.

I am going to rewrite this function to read the rows, create a struct, then return the struct that should resolve all the concerns brought up. The only reason I did not go this route before was solely the fact all the row parsling logic existed so it cut down on the amount of refactoring needed however I agree a refactor could be cleaner.

Reply:
@trentwatt actually upon further investigation I no longer want to perform this refactor in the current PR: https://github.com/determined-ai/determined/pull/5946#issuecomment-1424921223

Reply:
we don't need to do a refactor I don't think. it looks like every time we do 
```
rows, err = a.m.db.TrainingMetricsSeries(...)
```
or 
```
rows, err = a.m.db.ValidationMetricsSeries(...)
```
we do 
```
a.scanMetricsSeries(rows)
```

so we can just move the call to `scanMetricsSeries(rows)` to be inside `TrainingMetricsSeries` and `ValidationMetricsSeries` and have the return type of `TrainingMetricsSeries` be that of `scanMetricsSeries(rows)`.

that way `TrainingMetricsSeries` can `defer rows.close()` and  does not implicitly depend on the calling function to clean up.

Reply:
@trentwatt I had explicitly tried to remove the `scanMetricsSeries` call from the `Validation` and `Training` SeriesMetrics methods in an effort to separate DB query and business logic. I think the current implementation is a good middle ground. 

I implemented a struct which contains an `AverageMetrics` map which allows us to return all data not just just the 3 predefined `batches`, `time`, and `epoch` `[]lttb.Point` point arrays. This allow the callers of the `SeriesMetrics` methods to handle the data as they see fit.

Reply:
I am happy with the current implementation. 

My concern was just that I didn't want to have a function depend on the code that calls it to prevent memory leaks, but that looks to be resolved.